policy/modules/system/systemd.if

```diff
@@ -58,28 +69,603 @@ template(`systemd_role_template',`
 	# This domain is per-role because of the below transitions.
 	# See the systemd --user section of systemd.te for the
 	# remainder of the rules.
-	allow $1_systemd_t $3:process { setsched rlimitinh };
+	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
+	allow $1_systemd_t self:netlink_kobject_uevent_socket create_socket_perms;
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
-	allow $1_systemd_t self:process signal;
+	allow $1_systemd_t self:process { getsched signal };
 
 	files_search_home($1_systemd_t)
 
+	init_link_key_template($1_systemd_t)
+
+	miscfiles_watch_localization($1_systemd_t)
+
+	mount_read_runtime_files($1_systemd_t)
+	mount_watch_runtime_files($1_systemd_t)
+	mount_watch_reads_runtime_files($1_systemd_t)
+
+	allow $3 $1_systemd_t:process signal;
 	# Allow using file descriptors for user environment generators
 	allow $3 $1_systemd_t:fd use;
 	allow $3 $1_systemd_t:fifo_file rw_inherited_fifo_file_perms;
+	allow $3 $1_systemd_t:unix_stream_socket { connectto rw_socket_perms };
+
+	allow $1_systemd_t self:unix_stream_socket { connectto rw_socket_perms };
+
+	allow $1_systemd_t self:dbus { acquire_svc send_msg };
+	allow $3 $1_systemd_t:dbus send_msg;
+	allow $1_systemd_t $3:dbus send_msg;
+
+	allow $3 $1_systemd_t:system { disable enable reload start stop status };
 
 	# systemctl --user
 	stream_connect_pattern($3, systemd_user_runtime_t, systemd_user_runtime_t, $1_systemd_t)
 
+	search_dirs_pattern($1_systemd_t, $3, $3)
+	read_files_pattern($1_systemd_t, $3, $3)
+
 	can_exec($3, { systemd_run_exec_t systemd_analyze_exec_t })
 
+	dbus_exec($1_systemd_t)
```

Comment:
you should transition to $1_dbusd_t

Reply:
And then if you transition do $1_dbus_t then the whole scenario changes and you might be seeing less avc denials.

When you start to write a policy. always address transitions first. because else you might find your self needing to redo parts of it because it affects the policy

Reply:
Whoops, I completely forgot about this. Taking a look at this first.

Reply:
I made this change and am validating the effects of this. Unless I am doing something wrong, I haven't been able to see a lot of changes from here. Whenever I do `systemctl --user start myapp.service`, it doesn't look like my session dbus is running the service at all, but rather, the `systemd --user` instance is, so I'm not sure if this changes much.

Reply:
the dbus --user instance is probably socket activated. This means that when you log-in systemd --user will create the /run/user/*/bus sock file on dbus behalf and listens on it for connections. (verify that with systemctl --user status dbis.socket and ls -alZ $XDG_RUNTIME_DIR/bus

To make systemd start the dbus --user instance on demand you can simple run busctl --user. To verify whether that made systemd --user start dbus --user with the proper domain transitiion run ps xZ| grep dbus

Reply:
also ensure that systemd --user created the $XDG_RUNTIME_DIR/bus sock file with the proper context (something like dbusd_user_runtime_t?)

Reply:
After following your instructions, I end up with `dbus-daemon --session` running as `staff_dbusd_t` with a PPID of that of the `systemd --user` instance. The created dbus socket at `$XDG_RUNTIME_DIR/bus` has a type of `session_dbusd_runtime_t`. Does that seem right?